### ISSUE TITLE
The order of elements in `security_group_ids` has changed after Terraform apply

### ISSUE THREAD

**Issue by lyu571** [tencentcloudstack/terraform-provider-tencentcloud#2083]
### Affected Version
 \>=[v1.78.6](https://github.com/tencentcloudstack/terraform-provider-tencentcloud/releases/tag/v1.78.6)

### Affected Resource(s)

* tencentcloud_kubernetes_node_pool

### Terraform Configuration Files

```hcl
resource "tencentcloud_kubernetes_node_pool" "example_np" {
  name                     = "tf_example_node_pool"
  cluster_id               = tencentcloud_kubernetes_cluster.example_np.id
  max_size                 = 6              # set the node scaling range [1,6]
  min_size                 = 1
  vpc_id                   = local.first_vpc_id
  subnet_ids               = [local.first_subnet_id]
  retry_policy             = "INCREMENTAL_INTERVALS"
  desired_capacity         = 4
  enable_auto_scale        = true
  multi_zone_subnet_policy = "EQUALITY"
  tags = {
    "createdBy" = "terraform"
  }

  auto_scaling_config {
    instance_type              = var.default_instance_type
    system_disk_type           = "CLOUD_PREMIUM"
    system_disk_size           = "60"
    security_group_ids = ["sg-one", "sg-two", "sg-three"]

    data_disk {
      disk_type = "CLOUD_PREMIUM"
      disk_size = 50
    }
...
}
```

### Debug Output
<img width="1739" alt="image" src="https://github.com/tencentcloudstack/terraform-provider-tencentcloud/assets/13725630/78174d17-5a9e-4f40-a62c-700b994fcc5b">

### Expected Behavior

Create/Update security groups by the order in my tf code.

### Actual Behavior
The order of the security groups created does not follow the `security_group_ids` in tf code actually.
In other words, the order of elements in `security_group_ids` has changed after `terraform apply`.

### Steps to Reproduce

1. `terraform apply`
2. Compare the `security_group_ids` between tf code and TencentCloud Console/terraform.log.

### Important Factoids

None.

### References

https://registry.terraform.io/providers/tencentcloudstack/tencentcloud/latest/docs/resources/kubernetes_node_pool

**Comment by lyu571** [comment]
### Analysis:
1. The security group in TencentCloud is an ordered resource.
2. The provider changed `security_group_ids` to unordered to fix another bug(#1321) reported by TencentCloud AS product. However, we ignore the limitation introduced by the security group we mentioned before.

### RootCause:
A `TypeSet` field is misused for order-sensitive security groups which is caused by #1321.

**Comment by lyu571** [comment]
### Potential Solution:
#### Solution A（Better）:
Import a new field `orderly_security_group_ids` which can provide an ordered security group to the user, in the meanwhile, we keep the `security_group_ids` field to be compatible with the mass of existing resources previously.

#### Solution B:
keep `security_group_ids` but modify the type from `TypeSet` to `TypeList`.
* Affection: This solution may affect a variety of existing resources created and worked well previously, which is not reported any issue by the user.

**Comment by lyu571** [comment]
### Release:
https://github.com/tencentcloudstack/terraform-provider-tencentcloud/releases/tag/v1.81.22
